core/codec/cbor/cbor_decode_stream.hpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_CODEC_CBOR_CBOR_DECODE_STREAM_HPP
+#define CPP_FILECOIN_CORE_CODEC_CBOR_CBOR_DECODE_STREAM_HPP
+
+#include "codec/cbor/cbor_common.hpp"
+
+#include <cbor.h>
+#include <gsl/span>
+#include <vector>
+
+namespace fc::codec::cbor {
+  class CborDecodeStream {
+   public:
+    static constexpr auto is_cbor_decoder_stream = true;
+
+    explicit CborDecodeStream(gsl::span<const uint8_t> data);
+
+    template <typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
+    CborDecodeStream &operator>>(T &num) {
```

Comment:
As it has been found out with SCALE, it is nice to have a **decode<T>** function to avoid creating an object before decoding it when you can use it on-fly, without a separate object.